drivers/include/periph/spi.h

```diff
@@ -1,192 +1,174 @@
- *
- * @return              0 on success
- * @return              -1 on error
+ * The actual speed of the bus can vary to some extend, as the combination of
+ * CPU clock and available prescaler values on certain platforms may not make
+ * the exact values possible.
  */
-int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char data));
+#ifndef HAVE_SPI_CLK_T
+typedef enum {
+    SPI_CLK_100KHZ,         /**< drive the SPI bus with 100KHz */
+    SPI_CLK_400KHZ,         /**< drive the SPI bus with 400KHz */
+    SPI_CLK_1MHZ,           /**< drive the SPI bus with 1MHz */
+    SPI_CLK_5MHZ,           /**< drive the SPI bus with 5MHz */
+    SPI_CLK_10MHZ           /**< drive the SPI bus with 10MHz */
+} spi_clk_t;
+#endif
 
 /**
- * @brief Configure SCK, MISO and MOSI pins for the given SPI device
+ * @brief   Initialize pins used by the given SPI device
  *
- * @param[in] dev       SPI device to use
+ * @param[in] dev       initialize pins for this SPI device
+ * @param[in] cs        also initialize this chip select pin
  *
  * @return              0 on success
- * @return              -1 on error
+ * @return              -1 on invalid device
+ * @return              -2 on invalid CS
  */
-int spi_conf_pins(spi_t dev);
+int spi_init(spi_t dev, spi_cs_t cs);
 
 /**
- * @brief Get mutually exclusive access to the given SPI bus
+ * @brief   Start a new SPI transaction
  *
- * In case the SPI device is busy, this function will block until the bus is free again.
+ * Starting a new SPI transaction will get exclusive access to the SPI bus
+ * and configure it according to the given values. If another SPI transaction
+ * is active when this function is called, this function will block until the
+ * other transaction is complete (spi_relase was called).
  *
  * @param[in] dev       SPI device to access
+ * @param[in] mode      mode to use for the new transaction
+ * @param[in] clk       bus clock speed to use for the transaction
+ * @param[in] cs        chip select pin to use
  *
  * @return              0 on success
  * @return              -1 on error
  */
-int spi_acquire(spi_t dev);
+int spi_acquire(spi_t dev, spi_mode_t mode, spi_clk_t clk, spi_cs_t cs);
 
 /**
  * @brief Release the given SPI device to be used by others
  *
  * @param[in] dev       SPI device to release
- *
- * @return              0 on success
- * @return              -1 on error
  */
-int spi_release(spi_t dev);
+void spi_release(spi_t dev);
 
 /**
  * @brief Transfer one byte on the given SPI bus
  *
  * @param[in] dev       SPI device to use
  * @param[in] out       Byte to send out, set NULL if only receiving
- * @param[out] in       Byte to read, set NULL if only sending
  *
- * @return              Number of bytes that were transfered
- * @return              -1 on error
+ * @return              The byte that was received
  */
-int spi_transfer_byte(spi_t dev, char out, char *in);
+uint8_t spi_transfer_byte(spi_t bus, spi_cs_t cs, bool cont, uint8_t out);
```

Comment:
Is there a reason for not renaming "dev" to "bus" in spi_acquire and spi_release?
Also, it would help to understand the interface changes if the doxygen is updated together with the code